src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4869,7 +4869,7 @@ private static List<CompletionResult> GetDefaultProviderResults(
 
                 if (childNameList.Count == 0)
                 {
-                    return results;
+                    continue;
```

Comment:
Does the code path work for FileSystem provider? Has it the issue too?

Reply:
No. There's 2 separate provider completer methods: One for the filesystem provider, and a generic one for the others. I only made this mistake in the generic one.